tests/test_idpi/test_flexpart.py

```diff
@@ -64,59 +56,55 @@ def test_flexpart(data_dir, fieldextra):
     for f in ("FIS", "FR_LAND", "SDOR"):
         fs_ds_o[f] = fs_ds[f].isel(y_1=slice(None, None, -1))
 
-    assert_allclose(fs_ds_o["FIS"], ds["FIS"], rtol=3e-7, atol=5e-7)
-    assert_allclose(fs_ds_o["FR_LAND"], ds["FR_LAND"], rtol=3e-7, atol=5e-7)
-    assert_allclose(fs_ds_o["SDOR"], ds["SDOR"], rtol=3e-7, atol=5e-7)
-
-    # Compute few steps of a 3 hourly data
-    steps = 1, 2
-    for step, fs_ds in zip(steps, fs_ds_h):
-        fs_ds_o = {}
+    assert_allclose(fs_ds_o["FIS"], ds["FIS"].squeeze(), rtol=3e-7, atol=5e-7)
+    assert_allclose(fs_ds_o["FR_LAND"], ds["FR_LAND"].squeeze(), rtol=3e-7, atol=5e-7)
+    assert_allclose(fs_ds_o["SDOR"], ds["SDOR"].squeeze(), rtol=3e-7, atol=5e-7)
 
-        # Invert the latitude order in FX netcdf
-        for f in (
-            "U",
-            "V",
-            "T",
-            "QV",
-            "PS",
-            "U_10M",
-            "V_10M",
-            "T_2M",
-            "TD_2M",
-            "CLCT",
-            "W_SNOW",
-            "TOT_CON",
-            "TOT_GSP",
-            "SSR",
-            "SSHF",
-            "EWSS",
-            "ETADOT",
-        ):
-            fs_ds_o[f] = fs_ds[f].isel(y_1=slice(None, None, -1))
-
-        ds_out = flx.fflexpart(ds, step)
-
-        assert_allclose(
-            fs_ds_o["ETADOT"].transpose("y_1", "x_1", "z_1", "time").isel(time=0),
-            ds_out["OMEGA"],
-            rtol=3e-6,
-            atol=5e-5,
+    for field in (
```

Comment:
just looking at this long list, is there value in having a configurable way to say which fields are supposed to be present in an output dataset of a test and what the respective absolute and relative tolerances should be? We did something like this in the past [here](https://github.com/ai2cm/pace/blob/main/tests/savepoint/thresholds/fv_dynamics.yaml). But I'm open. This would also exceed the scope of this PR but might be worth considering in the future?

Reply:
I'm not too sure about this. I think having the list in the test is better than having to look it up somewhere else. However, the output field list has a few modifications with respect to the input field list. Maybe it could be defined as a diff to the input field list rather than a list on its own? Maybe @cosunae has more insight as the original author of the test.

Reply:
I see your point about tests being as self contained as possible.

For me the question is more on the maintainability side. Is it going to be easy in the future if operators get added  / changed to make sure the list is up to date and the two places and you're covering everything still.

In an ideal world you would loop over all the fields of the dataset and check. Now we have the issue of time dependent vs time independent variables. And then we have the issue of custom thresholds. I am just a bit worried about the pattern here, but again, totally out of scope

Reply:
I like the idea of getting the list of output fields from the output file generated from fieldextra. That should work, I think. we can also distinguish time dependent vs time independent variables from the dimensions. Originally I thought about parsing the fieldextra namelist, but getting the list from the output file sounds better choice. if we do it, I think we should apply the pattern to all tests. About the thresholds, a strategy where we apply a common threshold for all variables I think would be enough for testing and provide a simple strategy. Too tight thresholds for each variable might lead to a situation where we need to often relax them for many tests when we change data, compiler, etc. 
For cases where the thresholds need to be relaxed too much, we could specify custom ones in a external conf file (as suggested above).

Reply:
https://meteoswiss.atlassian.net/browse/APNRZ-133